crates/uv/tests/common/mod.rs

```diff
@@ -156,22 +161,47 @@ impl TestContext {
         result
     }
 
+    pub fn new_with_versions(python_versions: &[&str]) -> Self {
+        let mut context = Self::new(
+            python_versions
+                .first()
+                .expect("At least one test Python version must be provided"),
+        );
+
+        let python_versions: Vec<_> = python_versions
+            .iter()
+            .map(|version| PythonVersion::from_str(version).unwrap())
+            .zip(
+                python_toolchains_for_versions(&context.temp_dir, python_versions)
+                    .expect("Failed to find test Python versions"),
+            )
+            .collect();
+
+        for (version, path) in &python_versions {
+            context.add_filters_for_python_version(version, path.clone());
+        }
+
+        context.python_versions = python_versions;
+
+        context
+    }
+
     fn add_filters_for_python_version(&mut self, version: &PythonVersion, executable: PathBuf) {
         // Add filtering for the interpreter path
-        self.filters.extend(
-            Self::path_patterns(executable)
-                .into_iter()
-                .map(|pattern| (format!("{pattern}python.*"), format!("[PYTHON-{version}]"))),
-        );
+        for pattern in Self::path_patterns(executable) {
+            self.filters
+                .push_front((format!("{pattern}python.*"), format!("[PYTHON-{version}]")));
```

Comment:
These need to be in the front or they can get clobbered by other filters. Prompted the change to a double ended data structure.